firestore/src/android/aggregate_query_android.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIREBASE_FIRESTORE_SRC_ANDROID_AGGREGATE_QUERY_ANDROID_H_
+#define FIREBASE_FIRESTORE_SRC_ANDROID_AGGREGATE_QUERY_ANDROID_H_
+
+#include "firestore/src/android/promise_factory_android.h"
+#include "firestore/src/android/query_android.h"
+#include "firestore/src/android/wrapper.h"
+#include "firestore/src/include/firebase/firestore/aggregate_source.h"
+
+namespace firebase {
+namespace firestore {
+
+class AggregateQueryInternal : public Wrapper {
+ public:
+  // Each API of AggregateQuery that returns a Future needs to define an enum
+  // value here. For example, a Future-returning method Foo() relies on the enum
+  // value kFoo. The enum values are used to identify and manage Future in the
+  // Firestore Future manager.
+  enum class AsyncFn {
+    kGet = 0,
+    kCount,  // Must be the last enum value.
+  };
+
+  static void Initialize(jni::Loader& loader);
+
+  AggregateQueryInternal(FirestoreInternal* firestore,
+                         const jni::Object& object)
+      : Wrapper(firestore, object), promises_(firestore) {}
+
+  /**
+   * @brief Returns the query whose aggregations will be calculated by this
+   * object.
+   */
+  Query query();
+
+  /**
+   * @brief Executes the aggregate query and returns the results as a
+   * AggregateQuerySnapshot.
+   *
+   * @param[in] aggregate_source A value to configure the get behavior.
+   *
+   * @return A Future that will be resolved with the results of the
+   * AggregateQuery.
+   */
+  Future<AggregateQuerySnapshot> Get(AggregateSource aggregate_source);
```

Comment:
Get() can be const

Reply:
Should this be done in Query class as well? https://github.com/firebase/firebase-cpp-sdk/blob/044da556bc04404f10f6a0e247a676939acaf8bf/firestore/src/android/query_android.h#L361-L361

Reply:
Yes probably. Honestly, it's not a big deal because we don't apply `const` consistently throughout these Android wrapper classes. But I figured since you're writing new code we may as well apply the `const` in your new code.

Reply:
Actually I can't, since the implementation relies on some future logic that is not const.

https://github.com/firebase/firebase-cpp-sdk/blob/044da556bc04404f10f6a0e247a676939acaf8bf/firestore/src/android/promise_factory_android.h#L82-L82